### PR TITLE
3.7.0: Adds circuit breaking to PATCH,PUT,POST

### PR DESCRIPTION
**Pull request by smnalex**
* adds circuit breaker to PATCH, POST, PUT requests

### CODE REVIEW

**Review by laszlorozsas — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -36,22 +35,65 @@ jobs:
...
+            unset RACK_ENV &&
+            unset RAILS_ENV &&
+            bundle exec rspec
+
+      - save_cache:
+          key: bundle-2.7.4-rails_6-{{ .Branch }}
+          paths:
+            - ~/project/vendor/bundle
+      - save_cache:
+          key: bundle-2.7.0-rails_6
```
Comment (kouno) [inline .circleci/config.yml]: Keys don't match (`2.7.4`)

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,12 @@
 
 _A description of your awesome changes here!_
 
+### 3.7.0
+
+Features:
+
+- Adds circuit breakeing to PATCH PUT POST requests (#66)
```
Comment (kouno) [inline CHANGELOG.md]: `Adds circuit breaker to`

File: spec/routemaster/api_client_spec.rb [hunk 3]
```diff
@@ -109,10 +109,38 @@ def dummy; true; end
...
+        .with('ROUTEMASTER_CIRCUIT_BREAKER_ERROR_THRESHOLD', 50).and_return('1')
+      allow(ENV)
+        .to receive(:fetch)
+        .with('ROUTEMASTER_CIRCUIT_BREAKER_VOLUME_THRESHOLD', 50).and_return('1')
+
+      stub_request(method, url).to_timeout
+    end
+
+    it 'trips after the second request' do
+      5.times { expect { subject }.to raise_error StandardError }
```
Comment (kouno) [inline spec/routemaster/api_client_spec.rb:133]: Isn't subject memoized? I think you'll have to define a new method for that to work.

Comment (smnalex) [reply spec/routemaster/api_client_spec.rb:133]: no, the subject is evaluated each time.

**Review by kouno — APPROVED** [review]

**Pull request merged**